sdk/core/src/services/AffinidiVaultStorageService.ts

```diff
@@ -0,0 +1,184 @@
+    const privateKeyBuffer = this._keysService.getOwnPrivateKey()
+    const encryptedCredentials: VaultCredential[] = []
+
+    for (const credential of credentials) {
+      let credentialId = credential?.id
+      if (!isW3cCredential(credential)) {
+        credentialId = credential?.data?.id
+      }
+
+      const credentialIdHash = await this._platformEncryptionTools.computePersonalHash(privateKeyBuffer, credentialId)
+
+      const typeHashes = []
+      if (isW3cCredential(credential)) {
+        for (const credentialType of credential.type) {
+          const typeHash = await this._platformEncryptionTools.computePersonalHash(privateKeyBuffer, credentialType)
+          typeHashes.push(typeHash)
+        }
+      }
+
+      const cyphertext = await this._platformEncryptionTools.encryptByPublicKey(publicKeyBuffer, credential)
+
+      encryptedCredentials.push({
+        credentialId: credentialIdHash,
+        credentialTypes: typeHashes,
+        payload: cyphertext,
+      })
+    }
+
+    return encryptedCredentials
+  }
+
+  private async _computeTypesHashes(types: string[][]): Promise<string[][]> {
+    const privateKeyBuffer = this._keysService.getOwnPrivateKey()
+
+    const hashedTypes: string[][] = []
+    for (const subset of types) {
+      const hashedSubset: string[] = []
+      for (const type of subset) {
+        const hashedType = await this._platformEncryptionTools.computePersonalHash(privateKeyBuffer, type)
+        hashedSubset.push(hashedType)
+      }
+
+      hashedTypes.push(hashedSubset)
+    }
+
+    return hashedTypes
+  }
+
+  private async _decryptCredentials(encryptedCredentials: VaultCredential[]): Promise<any[]> {
+    const credentials: any[] = []
+
+    for (const encryptedCredential of encryptedCredentials) {
+      const credential = await this._decryptCredential(encryptedCredential)
+      credentials.push(credential)
+    }
+
+    return credentials
+  }
+
+  private async _decryptCredential(encryptedCredential: VaultCredential): Promise<any> {
+    const privateKeyBuffer = this._keysService.getOwnPrivateKey()
+
+    const credential = await this._platformEncryptionTools.decryptByPrivateKey(
+      privateKeyBuffer,
+      encryptedCredential.payload,
+    )
+
+    return credential
+  }
+
+  public async saveCredentials(credentials: any[], storageRegion: string): Promise<VaultCredential[]> {
+    const token = await this._authorizeVcVault()
+
+    const responses = []
+
+    const encryptedCredentials = await this._encryptCredentials(credentials)
+
+    for (const credential of encryptedCredentials) {
+      const { body } = await this._vaultApiService.storeCredential(token, storageRegion, credential.credentialId, {
+        credentialTypes: credential.credentialTypes,
+        payload: credential.payload,
+      })
+
+      responses.push(body)
+    }
+
+    return responses
+  }
+
+  public async searchCredentials(types: string[][], storageRegion: string): Promise<any[]> {
```

Comment:
sending an empty array to affinidi vault will cause it to return nothing as far as I remember that is how it was designed. you either need to send no `?types` parameter. or you need to supply a non empty array.

Reply:
I wonder why tests pass then.

This one, for example: https://github.com/affinityproject/affinidi-core-sdk/blob/yurii.s/FTL-1755/affinidi-vault-p1/sdk/node/test/integration/otp/index.test.ts#L100

Reply:
But overall I agree, empty array means result should be empty.

So even if it works now, we should make the code more clear, and pass `undefined` to this method when we want to get all credentials without filtering.

(Or `[[]]`, but IIRC affinidi vault explicitly disallows that)

Reply:
hm, okay I didn't know that and that is why put this check https://github.com/affinityproject/affinidi-core-sdk/blob/0cb982c366a90039c00c47639f80632c032ab7b1/common-libs/internal-api-clients/src/services/AffinidiVaultApiService.ts#L30. That is the reason why it is working